src/io.cr

```diff
@@ -776,6 +778,7 @@ abstract class IO
         end
 
         char2, char_bytesize2 = info2
+        bytes_read += char_bytesize2
```

Comment:
I don't think the exact number of bytes read needs to be computed, other than that it is non-zero. So `bytes_read` can be represented by a `Bool` instead and this line can be dropped

Reply:
Good point.
And actually I think it can be achieved even easier by replacing `break` in line 766 with `return false`. That's the only place where the method can return without having read anything. The last line would be `true`.

Reply:
@HertzDevil I've just pushed a second commit https://github.com/crystal-lang/crystal/pull/13882/commits/17752d184ce3dd793d2385519df2cc5b5ad9ab3c which tracks `bytes_read` as a `Bool` and drops the second accumulation `bytes_read += char_bytesize2` line as you've suggested.

@straight-shoota If I were to change line 766 from `break` to `return false`, I think this might incorrectly return false in any case where we've already read some bytes in previous iterations of the `while` loop and then hit EOF?